### PR TITLE
Add Kerberos authentication documentation

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: presto-docs/src/main/sphinx/admin/kerberos.rst [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+* ``--krb5-config-path``: Kerberos configuration file
+* ``--krb5-remote-service-name``: Presto coordinator Kerberos service name
+* ``--keystore-``: Path of keystore file to be used by HTTPS server
+* ``--keystore-password``: Keystore password
+
+Implementation access controller
+--------------------------------
+
+Additionally user need to provide implementation of access controller.
+This is done by implementing `SystemAccessControlFactory`` and ``SystemAccessControl`` interfaces.
```
Comment (kokosing) [inline presto-docs/src/main/sphinx/admin/kerberos.rst]: you are missing on ` in 

```
`SystemAccessControlFactory``
```

File: presto-docs/src/main/sphinx/admin/kerberos.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+Use implemented access controller
+---------------------------------
+
+To select and configure implement access controller ``access-control.properties`` configuration file must be added on Presto coordinator node.
+File must contain `access-control.name` property which should match name of implemented access controller.
+
+Other entries defined in configuration file are passed as map to ``SystemAccessControlFactory.crate(Map<String, String> config)`` method.
+
+
```
Comment (kokosing) [inline presto-docs/src/main/sphinx/admin/kerberos.rst]: remove trailing lines

File: presto-docs/src/main/sphinx/admin/kerberos.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
+===========================
+CLI Kerberos Authentication
+===========================
+
+Presto provides limited experimental support for Kerberos authentication when command-line interface is used.
```
Comment (rschlussel-zz) [inline presto-docs/src/main/sphinx/admin/kerberos.rst]: when the command-line

File: presto-docs/src/main/sphinx/admin/kerberos.rst [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+Presto coordinator node configuration
+-------------------------------------
+
+To use Kerberos based authentication following configuration properties must defined for Presto coordinator node:
```
Comment (rschlussel-zz) [inline presto-docs/src/main/sphinx/admin/kerberos.rst]: s/authentication following/authentication, the following/

Comment (rschlussel-zz) [inline presto-docs/src/main/sphinx/admin/kerberos.rst]: the Presto coordinator node

File: presto-docs/src/main/sphinx/admin/kerberos.rst [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    http.server.authentication.enabled=true
+    http.server.authentication.krb5.service-name=presto.coordinator.https
+    http.authentication.krb5.config=/etc/krb5.conf
+    http-server.https.enabled=true
+    http-server.https.port=7778
+    http-server.https.keystore.path=/etc/presto_keystore.jks
+    http-server.https.keystore.key=keystore_password
+
+* ``http.server.authentication.enabled``: Defines that Presto authentication should be enabled. Must be set to ``true``
+* ``http.server.authentication.krb5.service-name``: Defines Kerberos server name for Presto coordinator
```
Comment (rschlussel-zz) [inline presto-docs/src/main/sphinx/admin/kerberos.rst]: the Kerberos server name
the Presto coordinator

File: presto-docs/src/main/sphinx/admin/kerberos.rst [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+* ``http.authentication.krb5.config``: Kerberos configuration file
+* ``http-server.https.enabled``: Specifies if Presto server should listen for HTTPS
+* ``http-server.https.port``: HTTPS server port
+* ``http-server.https.keystore.path``: Path of keystore file to be used by HTTPS server
+* ``http-server.https.keystore.key``: Keystore password
+
+Presto CLI execution
+--------------------
+
+To use Kerberos on client side following extra parameters must be passed to presto-cli:
```
Comment (rschlussel-zz) [inline presto-docs/src/main/sphinx/admin/kerberos.rst]: the client side, the following

File: presto-docs/src/main/sphinx/admin/kerberos.rst [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    --keystore-path /etc/java_keystore.jks
+    --keystore-password the_keystore_password
+
+* ``--enable-authentication``: Defines that Presto authentication should be enabled.
+* ``--krb5-config-path``: Kerberos configuration file
+* ``--krb5-remote-service-name``: Presto coordinator Kerberos service name
+* ``--keystore-``: Path of keystore file to be used by HTTPS server
+* ``--keystore-password``: Keystore password
+
+Implementation access controller
```
Comment (rschlussel-zz) [inline presto-docs/src/main/sphinx/admin/kerberos.rst]: Access controller implementation

File: presto-docs/src/main/sphinx/admin/kerberos.rst [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+--------------------------------
+
+Additionally user need to provide implementation of access controller.
```
Comment (rschlussel-zz) [inline presto-docs/src/main/sphinx/admin/kerberos.rst]: the user needs

File: presto-docs/src/main/sphinx/admin/kerberos.rst (hunk 1 above)
Comment (rschlussel-zz) [inline presto-docs/src/main/sphinx/admin/kerberos.rst]: the ``SystemAccessControlFactory

File: presto-docs/src/main/sphinx/admin/kerberos.rst [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+This is done by implementing `SystemAccessControlFactory`` and ``SystemAccessControl`` interfaces.
+
+``SystemAccessControlFactory`` is responsible for creating ``SystemAccessControl`` instance and defining name later used
```
Comment (rschlussel-zz) [inline presto-docs/src/main/sphinx/admin/kerberos.rst]: a `SystemAccessControl` instance and defining a name

Also, it's a little unclear what kind of name it is defining

File: presto-docs/src/main/sphinx/admin/kerberos.rst [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+for access controller selection.
+
+``SystemAccessControl`` is responsible for:
+ * verifying if given Kerberos principal is authorized to execute queries as the specified user
```
Comment (rschlussel-zz) [inline presto-docs/src/main/sphinx/admin/kerberos.rst]: a given Kerberos principal

File: presto-docs/src/main/sphinx/admin/kerberos.rst [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+ * checking if given user can alter values for given system property
```
Comment (rschlussel-zz) [inline presto-docs/src/main/sphinx/admin/kerberos.rst]: a given user
a given system property

File: presto-docs/src/main/sphinx/admin/kerberos.rst [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+
+Implementation of ``SystemAccessControl`` and ``SystemAccessControlFactory`` must be wrapped as plugin and
```
Comment (rschlussel-zz) [inline presto-docs/src/main/sphinx/admin/kerberos.rst]: The implementation
a plugin
a Presto cluster

File: presto-docs/src/main/sphinx/admin/kerberos.rst [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+installed on Presto cluster.
+
+Use implemented access controller
```
Comment (rschlussel-zz) [inline presto-docs/src/main/sphinx/admin/kerberos.rst]: Using the implemented access controller

File: presto-docs/src/main/sphinx/admin/kerberos.rst [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+---------------------------------
+
+To select and configure implement access controller ``access-control.properties`` configuration file must be added on Presto coordinator node.
```
Comment (rschlussel-zz) [inline presto-docs/src/main/sphinx/admin/kerberos.rst]: This sentence was a bit confusing to me, but from what I think it is saying, it should be written as follows:

To select and configure the implemented access controller, the `access-controls.properties` configuration file must be added on the Presto coordinator node.

File: presto-docs/src/main/sphinx/admin/kerberos.rst [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+File must contain `access-control.name` property which should match name of implemented access controller.
```
Comment (rschlussel-zz) [inline presto-docs/src/main/sphinx/admin/kerberos.rst]: The file
the `access-control.name` property
the name of the implemented access controller

File: presto-docs/src/main/sphinx/admin/kerberos.rst [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+
+Other entries defined in configuration file are passed as map to ``SystemAccessControlFactory.crate(Map<String, String> config)`` method.
```
Comment (rschlussel-zz) [inline presto-docs/src/main/sphinx/admin/kerberos.rst]: the configuration file
a map
the ... method.

**Comment by rschlussel-zz** [comment]
Just a bunch of small grammatical changes (mostly adding 'the' and 'a') :).

File: presto-docs/src/main/sphinx/admin/kerberos.rst (hunk 8 above)
Comment (rschlussel-zz) [inline presto-docs/src/main/sphinx/admin/kerberos.rst]: an implementation

**Pull request closed without merging**